dll/shellext/netshell/lanstatusui.cpp

```diff
@@ -839,13 +840,15 @@ LANStatusDlg(
         case WM_INITDIALOG:
             pContext = (LANSTATUSUI_CONTEXT *)lParam;
             SetWindowLongPtr(hwndDlg, DWLP_USER, (LONG_PTR)lParam);
-            pContext->nIDEvent = SetTimer(hwndDlg, 0xFABC, 1000, NULL);
             return TRUE;
         case WM_TIMER:
             pContext = (LANSTATUSUI_CONTEXT*)GetWindowLongPtr(hwndDlg, DWLP_USER);
-            if (wParam == (WPARAM)pContext->nIDEvent)
+            if(pContext)
             {
-                UpdateLanStatus(pContext->hwndDlg, pContext);
+                if (wParam == (WPARAM)pContext->nIDEvent)
```

Comment:
Why add a new `if`?

Reply:
if(pContext) has been added for the sake of defensive programming since next call is a member of pContext, which we need to check non NULL. I prefer this to an ASSERT. OK for you ?